test/registry/DefaultRegistryClient.test.ts

```diff
@@ -40,10 +40,11 @@ describe('DefaultRegistryClient', () => {
       process.env.FPL_REGISTRY = 'https://custom-registry.example.org';
       new DefaultRegistryClient({ log: loggerSpy.log });
       new DefaultRegistryClient({ log: loggerSpy.log });
-      expect(loggerSpy.getLastMessage('info')).toBe(
-        'Using custom registry specified by FPL_REGISTRY environment variable: https://custom-registry.example.org'
-      );
-      expect(loggerSpy.getAllMessages('info')).toHaveLength(1);
+      // the hasLoggedCustomRegistry state is not reset so these expects fails
+      // expect(loggerSpy.getLastMessage('info')).toBe(
+      //   'Using custom registry specified by FPL_REGISTRY environment variable: https://custom-registry.example.org'
+      // );
+      // expect(loggerSpy.getAllMessages('info')).toHaveLength(1);
```

Comment:
As commented, the hasLoggedCustomRegistry state within the DefaultRegistryClient class is not being reset between tests or within test suite correctly

Reply:
Oh right... That `hasLoggedCustomRegistry` variable is in module scope -- which means the message is only logged once in the lifetime of the module (not once per instance of the `DefaultRegistryClient` class). It was probably logged while executing one of the other tests, so it won't be logged again. Let's just not do this test then. You can delete the test.